x-pack/plugins/cases/public/components/user_actions/description.tsx

```diff
@@ -0,0 +1,107 @@
+import { createCommonUserActionBuilder } from './common';
+import { UserActionUsername } from './username';
+import { UserActionAvatar } from './avatar';
+import { UserActionContentToolbar } from './content_toolbar';
+import { UserActionTimestamp } from './timestamp';
+import { UserActionMarkdown } from './markdown_form';
+import * as i18n from './translations';
+
+const DESCRIPTION_ID = 'description';
+
+const getLabelTitle = () => `${i18n.EDITED_FIELD} ${i18n.DESCRIPTION.toLowerCase()}`;
+
+type GetDescriptionUserActionArgs = Pick<
+  UserActionBuilderArgs,
+  | 'caseData'
+  | 'commentRefs'
+  | 'manageMarkdownEditIds'
+  | 'userCanCrud'
+  | 'handleManageMarkdownEditId'
+  | 'handleManageQuote'
+> &
+  Pick<UserActionTreeProps, 'onUpdateField' | 'isLoadingDescription'>;
+
+export const getDescriptionUserAction = ({
+  caseData,
+  commentRefs,
+  manageMarkdownEditIds,
+  isLoadingDescription,
+  userCanCrud,
+  onUpdateField,
+  handleManageMarkdownEditId,
+  handleManageQuote,
+}: GetDescriptionUserActionArgs): EuiCommentProps => {
+  return {
+    username: (
+      <UserActionUsername
+        username={caseData.createdBy.username}
+        fullName={caseData.createdBy.fullName}
+      />
+    ),
+    event: i18n.ADDED_DESCRIPTION,
+    'data-test-subj': 'description-action',
+    timestamp: <UserActionTimestamp createdAt={caseData.createdAt} />,
+    children: (
+      <UserActionMarkdown
+        ref={(element) => (commentRefs.current[DESCRIPTION_ID] = element)}
+        id={DESCRIPTION_ID}
+        content={caseData.description}
+        isEditable={manageMarkdownEditIds.includes(DESCRIPTION_ID)}
+        onSaveContent={(content: string) => {
+          onUpdateField({ key: DESCRIPTION_ID, value: content });
+        }}
+        onChangeEditable={handleManageMarkdownEditId}
+      />
+    ),
+    timelineIcon: (
+      <UserActionAvatar
+        username={caseData.createdBy.username}
+        fullName={caseData.createdBy.fullName}
+      />
+    ),
+    className: classNames({
+      isEdit: manageMarkdownEditIds.includes(DESCRIPTION_ID),
+    }),
+    actions: (
+      <UserActionContentToolbar
+        commentMarkdown={caseData.description}
+        id={DESCRIPTION_ID}
+        editLabel={i18n.EDIT_DESCRIPTION}
+        quoteLabel={i18n.QUOTE}
+        isLoading={isLoadingDescription}
+        onEdit={handleManageMarkdownEditId.bind(null, DESCRIPTION_ID)}
+        onQuote={handleManageQuote.bind(null, caseData.description)}
+        userCanCrud={userCanCrud}
+      />
+    ),
+  };
+};
+
+export const createDescriptionUserActionBuilder: UserActionBuilder = ({
+  userAction,
+  handleOutlineComment,
+}) => ({
+  build: () => {
+    const label = getLabelTitle();
+    const commonBuilder = createCommonUserActionBuilder({
+      userAction,
+      handleOutlineComment,
+      label,
+      icon: 'dot',
+    });
+
+    return [commonBuilder.build()];
```

Comment:
nit: seems like most of the callers to `createCommonUserActionBuilder` wrap the result in an array, do you think it'd make sense to have `createCommonUserActionBuilder` return an array of a single item?

Reply:
Yeap, I did that!